test/unit/utils/cuda/cuda_ptr_test.cpp

```diff
@@ -0,0 +1,222 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <iostream>
+#include <cassert>
+
+#include <cuda_runtime.h>
+#include <cuda.h>
+
+#include <cuda/cuda_utils.h>
+#include <nixl_log.h>
+
+using namespace std;
+
+#define CHECK_CUDA_ERROR(result, message)                                           \
+    do {                                                                            \
+        if (result != cudaSuccess) {                                                \
+            std::cerr << "CUDA: " << message << " (Error code: " << result          \
+                      << " - " << cudaGetErrorString(result) << ")" << std::endl;   \
+            exit(EXIT_FAILURE);                                                     \
+        }                                                                           \
+    } while(0)
+
+#define CHECK_CUDA_DRIVER_ERROR(result, message)                                    \
+    do {                                                                            \
+        if (result != CUDA_SUCCESS) {                                               \
+            const char *error_str;                                                  \
+            cuGetErrorString(result, &error_str);                                   \
+            std::cerr << "CUDA Driver: " << message << " (Error code: "             \
+                      << result << " - " << error_str << ")" << std::endl;          \
+            exit(EXIT_FAILURE);                                                     \
+        }                                                                           \
+    } while(0)
+
+void cudaQueryAddr(void *address, bool &is_dev,
```

Comment:
IMO cudaQueryAddr is redundant for logging since the test already knows how it allocates

Reply:
i would say the test should expect a specific type of memory and the nixl implementation should be doing the  query

Reply:
removed